subiquity/core.py

```diff
@@ -152,6 +159,28 @@ def __init__(self, opts, block_log_dir):
 
         self.install_confirmed = False
 
+    def subiquity_event(self, event):
+        if event["MESSAGE"] == "starting install":
+            if event["_PID"] == os.getpid():
+                return
+            if not self.install_lock_file.is_exclusively_locked():
+                return
+            from subiquity.ui.views.installprogress import (
+                InstallRunning,
+                )
+            tty = self.install_lock_file.read_content()
+            self.install_running = InstallRunning(self.ui.body, self, tty)
+            self.ui.body.show_stretchy_overlay(self.install_running)
+            schedule_task(self._hide_install_running())
+
+    async def _hide_install_running(self):
+        # Wait until the install has completed...
+        async with self.install_lock_file.shared():
+            pass
+        # And remove the overlay.
+        self.install_running = None
+        self.ui.body.remove_overlay()
+
```

Comment:
Any reason to not set state  within the lock context?

Reply:
No i guess not.